### PR TITLE
[DPE-3937] Add kyuubi_client provider interface

### PR DESCRIPTION
**Pull request by theoctober19th**

### CODE REVIEW

**Review by deusebio — APPROVED** [review]
Looks very good! well done!

File: tests/integration/app-charm/src/charm.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+# Copyright 2024 Ubuntu
```
Comment (welpaolo) [inline tests/integration/app-charm/src/charm.py:2]: Why do we have `Ubuntu` instead of `Canonical`?

**Review by welpaolo — APPROVED** [review]
LGTM

File: tests/integration/app-charm/src/charm.py (hunk 1 above)
Comment (theoctober19th) [reply tests/integration/app-charm/src/charm.py:2]: Good catch, thanks. Just updated.

**Pull request merged**